records.py

```diff
@@ -165,6 +165,11 @@ def dataset(self):
         # Create a new Tablib Dataset.
         data = tablib.Dataset()
 
+        # If the RecordCollection is empty, just return the empty set
+        # Check number of rows by typecasting to list
+        if len(list(self)) == 0:
```

Comment:
@timofurrer I can update that if needed. :blush: 
Would like to know what @kennethreitz thinks, if that's cool with you.